arangod/Cluster/ClusterInfo.cpp

```diff
@@ -3813,7 +3836,11 @@ Result ClusterInfo::createCollectionsCoordinator(
       }
     }
 
-    if (nrDone->load(std::memory_order_acquire) == infos.size() &&
+    auto nrDone =
+        report->doUnderLock([](CreateCollectionReport const& report) -> size_t {
+          return report.nrDone;
+        });
```

Comment:
It was atomic, why do you decide to change it to mutex?

Reply:
Simplicity.